core/interfaces/i_plugin.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a plugin.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IPlugin');
+
+
+/**
+ * @interface
+ */
+Blockly.IPlugin = function() {};
+
+/**
+ * Initializes the plugin.
+ * @return {void}
+ */
+Blockly.IPlugin.prototype.init;
```

Comment:
Many UI plugin candidates (eg: trashcan, zoomcontrols) have both init and createDom. We should try and formalize when to use which and create a subtype of UIPlugin that implements createDom.